### DISCUSSION TITLE
Usage scenarios for teams

### DISCUSSION THREAD

**Discussion by davinishi** [thunderclient/thunder-client-support#608] [Q&A]
Hi,

I was looking forward for a http client capable to integrate with git which would enhance sharing capabilities across teams.

After investing some time configuring it, this is my concept:

- The client is connected to a single git repository;
- Different sets of collections are created within branches; 
- Each branch contains collections that corresponds to a specific system or similar systems;
- In this scenario, the branches are independent.

This concept works great for what I was looking for, but there is one minor problem, I would like to have one master branch aggregating all the collections but not losing the ease of having the children branches containing independent sets of collections.
Updating one child branch, would affect the master branch but not the other children branches.

Q1: Is this possible?
Q2: Is this achievable using git submodules? (I tried but couldn't make it work)
Q3: Is there a better way to achive the same or better result?
Q4: Was the "Workspace Relative Path" created for this?
Q5: Can we have a more detailed tutorial on how to configure "Workspace Relative Path"? I cracked my head trying to figure out how to use it.

**Comment by rangav** [comment]
Hi @davinishi 

You can use `Save To Workspace` setting, that will save the requests data in current git project.

If you have branches with different collections data, then the extension will load from that branch, assuming the requests data is stored in root of the branch.

When you change the branch you need to manually refresh the sidebar data with reload data icon
![image](https://user-images.githubusercontent.com/8637550/171408101-952cecac-9c5a-4482-849c-68be803e1226.png)

`Workspace Relative Path` is created for big projects that have sub-projects. where requests data is stored in different location instead of project root. you can use it if this use-case suits you.

Let me know if you have any further questions

**Reply by davinishi** [reply]
@rangav 
The hints you gave regarding saving and manually refreshing I was already familiar, it is not a big deal.
My question is actually based on how to work with multiple branches, for instance:

-> master (collection 1, collection 2, collection 3)
--> branch_a (collection 1)
--> branch_b (collection 2)
--> branch_c (collection 3)

I don't want to merge branches, instead, I want to commit to branch_a and commit to master.

I tried to use submodules, but the problem is that I couldn't find a way to use tests folder other than 'thunder-tests' hence the commit always shows conflict with master since it has the same folder name. Maybe I just want to find how to set a different root folder.

**Reply by rangav** [reply]
When you change branch to branch_a the extension will load collection 1 data(click manual reload button after branch change)

did you try that?

**Reply by davinishi** [reply]
@rangav 

Yep, and it is working fine ;)

The problem is that I would like to have one main branch where all the collections are saved, this way I will have one branch where I can see all the collections and other small branches with specific collections.

Perhaps this is not a good way to manage things, but in my case scenario it does make sense.

**Reply by rangav** [reply]
The use case you mentioned above will work. Let me know if you have any further questions
